data/extension/init/templates/tinygo/envoy.filters.network/default/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+var (
+	connectionCounterName = "proxy_wasm_go.connection_counter"
+	counter               proxywasm.MetricCounter
+)
+
+type context struct{ proxywasm.DefaultContext }
+
+func main() {
+	proxywasm.SetNewRootContext(func(contextID uint32) proxywasm.RootContext { return context{} })
+	proxywasm.SetNewStreamContext(func(contextID uint32) proxywasm.StreamContext { return context{} })
+}
+
+func (ctx context) OnVMStart(int) bool {
+	var err error
+	counter, err = proxywasm.DefineCounterMetric(connectionCounterName)
+	if err != nil {
+		proxywasm.LogCritical("failed to initialize connection counter: ", err.Error())
+	}
+	return true
+}
+
+func (ctx context) OnNewConnection() types.Action {
```

Comment:
Let's use `func (ctx *context)` version to clearly indicate that `context` is supposed to have mutable state.